server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader9to10.java

```diff
@@ -738,33 +739,36 @@ static Path resolveRelativeDelete(String oldDelete, String upgradeProperty) {
   /**
    * Remove old temporary map files to prevent problems during recovery.
    */
-  static void dropSortedMapWALFiles(VolumeManager vm) {
-    Path recoveryDir = new Path("/accumulo/recovery");
-    try {
-      if (!vm.exists(recoveryDir)) {
-        log.info("There are no recovery files in /accumulo/recovery");
-        return;
-      }
-      List<Path> directoriesToDrop = new ArrayList<>();
-      for (FileStatus walDir : vm.listStatus(recoveryDir)) {
-        // map files will be in a directory starting with "part"
-        Path walDirPath = walDir.getPath();
-        for (FileStatus dirOrFile : vm.listStatus(walDirPath)) {
-          if (dirOrFile.isDirectory()) {
-            directoriesToDrop.add(walDirPath);
-            break;
+  static void dropSortedMapWALFiles(ServerContext context) {
+    VolumeManager vm = context.getVolumeManager();
+    for (Volume volume : vm.getVolumes()) {
+      Path recoveryDir = volume.prefixChild("/accumulo/recovery");
```

Comment:
I suspect `/accumulo` is not needed as it will be [part of the volume](https://github.com/apache/accumulo/blob/7db0561cac3d42ba09d1fb038dad1ff1a03afc46/assemble/conf/accumulo.properties#L24).  Also there is constant `RECOVERY_DIR` in ServerConstants that could be used.

Reply:
Right, forgot about that. I think an IT would have caught that but unfortunately I think it's too involved to try and recreate. I will test with Uno one more time after we think the changes are solid.

Reply:
Even better, I think I can use the method in `ServerConstants` to get the directories. Check it out: https://github.com/apache/accumulo/blob/618ba7a9f88342ea7c8d692ce19df34d88bf07c8/server/base/src/main/java/org/apache/accumulo/server/ServerConstants.java#L169